src/scenic/syntax/compiler.py

```diff
@@ -476,6 +479,63 @@ def visit_MonitorDef(self, node: s.MonitorDef):
             body=node.body,
         )
 
+    def visit_ScenarioDef(self, node: s.ScenarioDef):
+        # TODO(shun): assert not in behavior or compose
+
+        # Set up arguments for setup and compose blocks
+        args: ast.arguments = self.visit(node.args)
+        args.posonlyargs = initialBehaviorArgs + args.posonlyargs
+
+        # Get preconditions and invariants
+        preconditions, invariants = self.separatePreconditionsAndInvariants(node.header)
+
+        setup = node.setup
+        compose = node.compose
+
+        # Find all locals of the scenario, which will be shared amongst the various blocks
+        allLocals = set()
+        if node.compose:
```

Comment:
Any reason not to just use `compose` and `setup` respectively for these kinds of conditions? This makes it look like we're saving the old value of `node.compose` in case it changes.

Reply:
My intention was to use `compose` and `setup` for compiled AST nodes and use `node.compose` and `node.setup` to access the original nodes, but I must have mixed them up.

f7e299bc671ddec259cac0e8de2e659732226085

This should fix the problems. Now `node.compose` and `node.setup` are compiled to `compose` and `setup`, respectively, and then will be used to create a class definition.